deptofdefense/move.mil
# Refactor Locator Maps

## Checklist

I have…


## Summary of Changes

This pull request refactors the Locator Maps code introduced in #89. The resulting user experience remains largely consistent. Users can perform searches by ZIP code and by their location with Transportation Offices ordered by distance from the ZIP code's center latitude/longitude or from the user's geolocation.

Data quality remains an issue but this will be refined in future updates pending receipt of cleaner data.

This PR also resolves #88.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout refactor-office-locator`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. run `bin/rails db:drop db:create db:migrate db:seed`,
1. run `bin/rails server`, and
1. load up http://localhost:3000/resources/locator-maps in your Web browser of choice,
1. try performing ZIP code searches and location searches.

## Screenshots

![search-initial](https://user-images.githubusercontent.com/27780860/31029954-a788d996-a521-11e7-899d-2eff17a8df4e.png)

![search-location-prompt](https://user-images.githubusercontent.com/27780860/31029965-acf03f64-a521-11e7-85d7-cab7ef7ab9b4.png)

![search-geolocation-error](https://user-images.githubusercontent.com/27780860/31029998-c0c800c6-a521-11e7-9de2-6af0c0239b21.png)

<table>
  <tr>
    <td><img src="https://user-images.githubusercontent.com/27780860/31030003-c45185fa-a521-11e7-8a23-7954e8244c6b.png" alt="coordinates-search-fullpage"></td>
    <td><img src="https://user-images.githubusercontent.com/27780860/31030005-c6e5d8e8-a521-11e7-93a0-b6377016bd4c.png" alt="zip-search-fullpage"></td>
  </tr>
</table>
